File: jajo-backend/src/main/java/jajobackend/model/Emporium.java
package jajobackend.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.Data;
import lombok.ToString;
import org.hibernate.annotations.Cascade;

import javax.persistence.*;
import java.util.List;

@Entity
@Data
public class Emporium {

    @Id
    @GeneratedValue (strategy = GenerationType.IDENTITY)
    @Column(name = "emporium_id")
    private Long id;

    private String name;

    private Boolean isSunday;

    //One to One
    @OneToOne(mappedBy = "emporium")
    @JsonIgnore
    @Cascade(org.hibernate.annotations.CascadeType.ALL)
    private ProductsIncome productsIncome;

    //One to Many
    @OneToMany (mappedBy = "emporium")
    @JsonIgnore
    @ToString.Exclude
    @Cascade(org.hibernate.annotations.CascadeType.ALL)
    private List<Transport> transports;

}

File: jajo-frontend/src/app/components/model/emporium.ts
import {ProductsIncome} from "./productsIncome";

export interface GetEmporium {
  id: number;
  name: string;
  isSunday: boolean;
  productsIncome: ProductsIncome;
}

export interface PostEmporium {
  id: number;
  name: string;
  isSunday: boolean;
  productsIncome: ProductsIncome;
}
